Dockerfile

```diff
@@ -1,41 +1,53 @@
-# Build the Go binary in a separate stage utilizing Makefile
+# Build stage
 FROM golang:1.22 AS builder
 
-# Install necessary packages for the final image
+WORKDIR /app
+
+# Install Node.js and Yarn
 RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
-    curl sudo gpg lsb-release software-properties-common \
+    curl gpg \
     && curl -fsSL https://deb.nodesource.com/setup_lts.x | bash - \
-    && curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | gpg --dearmor -o /usr/share/keyrings/yarn-archive-keyring.gpg \
-    && apt remove cmdtest \
-    && echo "deb [signed-by=/usr/share/keyrings/yarn-archive-keyring.gpg] https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list \
-    && apt-get update && apt-get install -y git yarn apt-utils
+    && apt-get install -y nodejs \
+    && npm install -g yarn
 
-WORKDIR /app
-COPY go.mod go.sum ./
-RUN go mod download
+# Copy the entire project
 COPY . .
 
+# Install contract dependencies
+RUN cd contracts && yarn install
```

Comment:
this shouldn't be required anymore as is being run here: https://github.com/masa-finance/masa-oracle/blob/609e2ecc7be7221bebb4be607392b84620fe7728/contracts/embed.go#L1

```suggestion
```